src/cluster/index.js

```diff
@@ -102,6 +104,29 @@ module.exports = class Cluster {
     return await this.brokerPool.findBroker({ nodeId })
   }
 
+  /**
+   * @public
+   * @returns {Promise<Broker>}
+   */
+  async findControllerBroker() {
+    const { metadata } = this.brokerPool
+
+    // controllerId is an int32, it's safe to cast to Number
+    if (!metadata || Number.isNaN(Number(metadata.controllerId))) {
```

Comment:
I would change this to `metadata.controllerId == null` instead. It's true that since it's an int32 it is safe to cast it, but it feels really wrong when all you really want to do is check if it's set. `== null` (note the loose comparison) will be true if it's undefined or null, but false if it's 0.

Reply:
Fixed here 31fdac162bab810f829b0dd392e3641d23727d20